pkg/webhooks/machine_webhook.go

```diff
@@ -1297,6 +1304,31 @@ func validateShieldedInstanceConfig(providerSpec *machinev1beta1.GCPMachineProvi
 	return errs
 }
 
+func validateGCPConfidentialComputing(providerSpec *machinev1beta1.GCPMachineProviderSpec) (errs []error) {
+	if providerSpec.ConfidentialCompute != "" && providerSpec.ConfidentialCompute != machinev1beta1.ConfidentialComputePolicyEnabled && providerSpec.ConfidentialCompute != machinev1beta1.ConfidentialComputePolicyDisabled {
```

Comment:
I thought I commented here but maybe it didn't save, I think this should be moved to a switch, it would be easier to reason about and evolve later right?

```
switch providerSpec.ConfidentialCompute {
case machinev1beta1.ConfidentialComputePolicyEnabled:
  // Check on host maintenance
case machinev1beta1.ConfidentialComputePolicyDisabled, "":
default:
  errs = ... // return invalid value error
}
```